src/H5S.c

```diff
@@ -1655,9 +1655,10 @@ H5S_decode(const unsigned char **p)
     if (H5S_select_all(ds, FALSE) < 0)
         HGOTO_ERROR(H5E_DATASPACE, H5E_CANTSET, NULL, "unable to set all selection")
 
-    /* Decode the select part of dataspace.  I believe this part always exists. */
+    /* Decode the select part of dataspace.  I believe this part always exists.
+     *  Because size of buffer is unknown, just assume arbitrarily large buffer to allow decoding. */
```

Comment:
Just a nitpick: I'm guessing at this point we can probably cleanup the part of the comment about this part always existing.